File: repository/src/test/java/by/rekuts/travelagency/repository/CountryRepositoryTest.java
package by.rekuts.travelagency.repository;

import by.rekuts.travelagency.domain.Country;
import by.rekuts.travelagency.repository.config.TestRepositoryConfig;
import lombok.extern.slf4j.Slf4j;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.transaction.annotation.Transactional;

@Slf4j
@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = TestRepositoryConfig.class)
@ActiveProfiles("testScope")
@Transactional
public class CountryRepositoryTest {

    @Autowired
    CountryRepository countryRepository;

    @Test
    public void insertCountryTestTrue() {
        var country = new Country();
        country.setName("Užupis");
        var specification = new CountrySpecification();
        var countFirst = countryRepository.getList(specification).size();
        countryRepository.insert(country);
        var countLast = countryRepository.getList(specification).size();
        Assert.assertEquals(1, countLast - countFirst);
    }

    @Test
    public void deleteCountryTest() {
        var specification = new CountrySpecification();
        specification.setId(230);
        var countFirst = countryRepository.getList(specification).size();
        countryRepository.delete(230);
        var countLast = countryRepository.getList(specification).size();
        Assert.assertEquals(1, countFirst - countLast);
    }

    @Test
    public void getCountryByIdTrue() {
        Assert.assertEquals("Albania", (countryRepository.getList(new CountrySpecification(2)).get(0)).getName().trim());
    }

    @Test
    public void getAllCountriesTest() {
        var specification = new CountrySpecification();
        var countries = countryRepository.getList(specification);
        Assert.assertEquals(199, countries.size());
    }
}

File: repository/src/test/scala/CountryRepositoryTest.scala
import java.util

import by.rekuts.travelagency.domain.Country
import by.rekuts.travelagency.repository.config.TestRepositoryConfig
import by.rekuts.travelagency.repository.{CountryRepository, CountrySpecification}
import org.junit.{Assert, Test}
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
import org.springframework.test.context.{ActiveProfiles, ContextConfiguration}
import org.springframework.transaction.annotation.Transactional

@RunWith(classOf[SpringJUnit4ClassRunner])
@ContextConfiguration(classes = Array(classOf[TestRepositoryConfig]))
@ActiveProfiles(Array("testScope"))
@Transactional
class CountryRepositoryTest {
  @Autowired
  private var countryRepository: CountryRepository = _

  @Test
  def insertCountryTestTrue(): Unit = {
    val country: Country = new Country
    country.name_$eq("Užupis")
    val specification: CountrySpecification = new CountrySpecification
    val countFirst: Int = countryRepository.getList(specification).size
    countryRepository.insert(country)
    val countLast: Int = countryRepository.getList(specification).size
    Assert.assertEquals(1, countLast - countFirst)
  }

  @Test def deleteCountryTest(): Unit = {
    val specification: CountrySpecification = new CountrySpecification
    specification.id_$eq(230)
    val countFirst: Int = countryRepository.getList(specification).size
    countryRepository.delete(230)
    val countLast: Int = countryRepository.getList(specification).size
    Assert.assertEquals(1, countFirst - countLast)
  }

  @Test def receiveCountryByIdTrue(): Unit = {
    Assert.assertEquals("Albania", countryRepository.getList(new CountrySpecification(2)).get(0).name.trim)
  }

  @Test def receiveAllCountriesTest(): Unit = {
    val specification: CountrySpecification = new CountrySpecification
    val countries: util.List[Country] = countryRepository.getList(specification)
    Assert.assertEquals(199, countries.size)
  }
}
